### ISSUE TITLE
Amplify.Auth.fetchAuthSession returns an expired tokenID

### ISSUE THREAD

**Issue by oantacamelia** [aws-amplify/amplify-flutter#2394]
### Description

We are using 
```
 final result = await Amplify.Auth.fetchAuthSession(
        options: CognitoSessionOptions(getAWSCredentials: true));
    final cognitoAuthSession = (result as CognitoAuthSession);
   return cognitoAuthSession.userPoolTokens?.idToken;
```
 to get the idToken to use it further with our API. The problem is that the idToken returned remains the same and never changes even if it's actually expired.

Our amplify setup is as following.
1. We used amplify CLI to init project
2. We used amplify import auth to add an existing User Pool only 

When we create a new account we see the following logs, but the logging in afterwards works and the user is added in the pool:
`W/CognitoUserSession(31560): CognitoUserSession is not valid because idToken is null.
D/AWSMobileClient(31560): Sending password.
D/AWSMobileClient(31560): Using USER_SRP_AUTH for flow type.
D/AWSMobileClient(31560): _federatedSignIn: Putting provider and token in store
W/AWSMobileClient(31560): Failed to federate tokens during sign-in
W/AWSMobileClient(31560): java.lang.Exception: Federation is not enabled, please check if you have CognitoIdentity configured.
W/AWSMobileClient(31560): 	at com.amazonaws.mobile.client.AWSMobileClient$11.run(AWSMobileClient.java:1882)
W/AWSMobileClient(31560): 	at com.amazonaws.mobile.client.internal.InternalCallback.await(InternalCallback.java:115)
W/AWSMobileClient(31560): 	at com.amazonaws.mobile.client.AWSMobileClient.federatedSignInWithoutAssigningState(AWSMobileClient.java:1835)
W/AWSMobileClient(31560): 	at com.amazonaws.mobile.client.AWSMobileClient$6$1.onSuccess(AWSMobileClient.java:1269)
W/AWSMobileClient(31560): 	at com.amazonaws.mobileconnectors.cognitoidentityprovider.CognitoUser$6.onSuccess(CognitoUser.java:1102)
W/AWSMobileClient(31560): 	at com.amazonaws.mobileconnectors.cognitoidentityprovider.CognitoUser$27.run(CognitoUser.java:3135)
W/AWSMobileClient(31560): 	at com.amazonaws.mobileconnectors.cognitoidentityprovider.CognitoUser$24.run(CognitoUser.java:3002)
W/AWSMobileClient(31560): 	at com.amazonaws.mobileconnectors.cognitoidentityprovider.continuations.AuthenticationContinuation.continueTask(AuthenticationContinuation.java:147)
W/AWSMobileClient(31560): 	at com.amazonaws.mobile.client.AWSMobileClient$6$1.getAuthenticationDetails(AWSMobileClient.java:1322)
W/AWSMobileClient(31560): 	at com.amazonaws.mobileconnectors.cognitoidentityprovider.CognitoUser.getSession(CognitoUser.java:1035)
W/AWSMobileClient(31560): 	at com.amazonaws.mobile.client.AWSMobileClient$6.run(AWSMobileClient.java:1254)
W/AWSMobileClient(31560): 	at com.amazonaws.mobile.client.internal.InternalCallback$1.run(InternalCallback.java:101)
W/AWSMobileClient(31560): 	at java.lang.Thread.run(Thread.java:920)
D/AWSMobileClient(31560): Inspecting user state details
D/AWSMobileClient(31560): hasFederatedToken: false provider: our link provider
`

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [X] Auth
- [ ] Authenticator
- [ ] DataStore
- [ ] Storage

### Steps to Reproduce

1. Create an amplify project and add an auth env
2. Login using email and password
3.  Use ` final result = await Amplify.Auth.fetchAuthSession(
        options: CognitoSessionOptions(getAWSCredentials: true));
    final cognitoAuthSession = (result as CognitoAuthSession);
   return cognitoAuthSession.userPoolTokens?.idToken;` to retrieve the idToken
4. Wait until token is expired (tried waiting in app and also with the app closed same result). Retrieve the token again as done above. The returned token is the same as the old one. 

### Platforms

- [ ] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Environment

```bash
[✓] Flutter (Channel stable, 3.3.6, on macOS 12.2 21D49 darwin-arm, locale en-RO)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
```

### Dependencies

```bash
Dart SDK 2.18.2
Flutter SDK 3.3.6
instagrid 1.0.0+1

dependencies:
- amplify_auth_cognito 0.6.10 [amplify_auth_cognito_android amplify_auth_cognito_ios amplify_core aws_common collection flutter meta plugin_platform_interface]
- amplify_flutter 0.6.10 [amplify_core amplify_datastore_plugin_interface amplify_flutter_android amplify_flutter_ios aws_common collection flutter meta plugin_platform_interface]
- carousel_slider 4.1.1 [flutter]
- cupertino_icons 1.0.5
- equatable 2.0.5 [collection meta]
- fl_chart 0.55.2 [flutter equatable]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_bloc 8.1.1 [flutter bloc provider]
- flutter_countdown_timer 4.1.0 [flutter]
- flutter_native_splash 2.2.14 [args flutter flutter_web_plugins js html image meta path universal_io xml yaml]
- flutter_secure_storage 6.0.0 [flutter flutter_secure_storage_linux flutter_secure_storage_macos flutter_secure_storage_platform_interface flutter_secure_storage_web flutter_secure_storage_windows meta]
- flutter_staggered_grid_view 0.6.2 [flutter]
- flutter_svg 1.1.6 [flutter meta path_drawing vector_math xml]
- geolocator 9.0.2 [flutter geolocator_platform_interface geolocator_android geolocator_apple geolocator_web geolocator_windows]
- google_fonts 3.0.1 [flutter http path_provider crypto]
- google_maps_flutter 2.2.1 [flutter google_maps_flutter_android google_maps_flutter_ios google_maps_flutter_platform_interface]
- maps_launcher 2.0.1 [flutter flutter_web_plugins url_launcher]
- percent_indicator 4.2.2 [flutter]
- provider 6.0.4 [collection flutter nested]
- qr_code_scanner 1.0.1 [js flutter flutter_web_plugins]
- shared_preferences 2.0.15 [flutter shared_preferences_android shared_preferences_ios shared_preferences_linux shared_preferences_macos shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- url_launcher 6.1.6 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]

transitive dependencies:
- amplify_auth_cognito_android 0.6.10 [flutter]
- amplify_auth_cognito_ios 0.6.10 [amplify_core flutter]
- amplify_core 0.6.10 [aws_common collection flutter intl json_annotation meta plugin_platform_interface uuid]
- amplify_datastore_plugin_interface 0.6.10 [amplify_core collection flutter meta]
- amplify_flutter_android 0.6.10 [flutter]
- amplify_flutter_ios 0.6.10 [amplify_core flutter]
- archive 3.3.4 [crypto path pointycastle]
- args 2.3.1
- async 2.9.0 [collection meta]
- aws_common 0.1.1 [async collection http meta stream_transform uuid]
- bloc 8.1.0 [meta]
- characters 1.2.1
- clock 1.1.1
- collection 1.16.0
- convert 3.1.1 [typed_data]
- crypto 3.0.2 [typed_data]
- csslib 0.17.2 [source_span]
- ffi 2.0.1
- file 6.1.4 [meta path]
- flutter_plugin_android_lifecycle 2.0.7 [flutter]
- flutter_secure_storage_linux 1.1.1 [flutter flutter_secure_storage_platform_interface]
- flutter_secure_storage_macos 1.1.1 [flutter flutter_secure_storage_platform_interface]
- flutter_secure_storage_platform_interface 1.0.0 [flutter plugin_platform_interface]
- flutter_secure_storage_web 1.0.2 [flutter flutter_web_plugins flutter_secure_storage_platform_interface js]
- flutter_secure_storage_windows 1.1.2 [flutter flutter_secure_storage_platform_interface]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- geolocator_android 4.1.4 [flutter geolocator_platform_interface]
- geolocator_apple 2.2.3 [flutter geolocator_platform_interface]
- geolocator_platform_interface 4.0.7 [flutter plugin_platform_interface vector_math meta]
- geolocator_web 2.1.6 [flutter flutter_web_plugins geolocator_platform_interface]
- geolocator_windows 0.1.1 [flutter geolocator_platform_interface]
- google_maps_flutter_android 2.3.3 [flutter flutter_plugin_android_lifecycle google_maps_flutter_platform_interface stream_transform]
- google_maps_flutter_ios 2.1.12 [flutter google_maps_flutter_platform_interface stream_transform]
- google_maps_flutter_platform_interface 2.2.4 [collection flutter plugin_platform_interface stream_transform]
- html 0.15.1 [csslib source_span]
- http 0.13.5 [async http_parser meta path]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- image 3.2.2 [archive meta xml]
- intl 0.17.0 [clock path]
- js 0.6.4
- json_annotation 4.7.0 [meta]
- material_color_utilities 0.1.5
- meta 1.8.0
- nested 1.0.0 [flutter]
- path 1.8.2
- path_drawing 1.0.1 [vector_math meta path_parsing flutter]
- path_parsing 1.0.1 [vector_math meta]
- path_provider 2.0.11 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.21 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.11 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.7 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.6 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.5 [flutter platform plugin_platform_interface]
- path_provider_windows 2.1.3 [ffi flutter path path_provider_platform_interface win32]
- petitparser 5.1.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.3 [meta]
- pointycastle 3.6.2 [collection convert js]
- process 4.2.4 [file path platform]
- shared_preferences_android 2.0.14 [flutter shared_preferences_platform_interface]
- shared_preferences_ios 2.1.1 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.1.1 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_macos 2.0.4 [flutter shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.1.0 [flutter plugin_platform_interface]
- shared_preferences_web 2.0.4 [flutter flutter_web_plugins shared_preferences_platform_interface]
- shared_preferences_windows 2.1.1 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- sky_engine 0.0.99
- source_span 1.9.0 [collection path term_glyph]
- stream_transform 2.1.0
- string_scanner 1.1.1 [source_span]
- term_glyph 1.2.1
- typed_data 1.3.1 [collection]
- universal_io 2.0.4 [collection crypto meta typed_data]
- url_launcher_android 6.0.21 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.1.1 [flutter plugin_platform_interface]
- url_launcher_web 2.0.13 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.1 [flutter url_launcher_platform_interface]
- uuid 3.0.6 [crypto]
- vector_math 2.1.2
- win32 3.1.1 [ffi]
- xdg_directories 0.2.0+2 [meta path process]
- xml 6.1.0 [collection meta petitparser]
- yaml 3.1.1 [collection source_span string_scanner]
```

### Device

Pixel 5

### OS

Android 12

### Deployment Method

Amplify CLI

### CLI Version

10.5.0

### Amplify Config

const amplifyconfig = ''' {
    "UserAgent": "aws-amplify-cli/2.0",
    "Version": "1.0",
    "auth": {
        "plugins": {
            "awsCognitoAuthPlugin": {
                "UserAgent": "aws-amplify-cli/0.1.0",
                "Version": "0.1.0",
                "IdentityManager": {
                    "Default": {}
                },
                "CognitoUserPool": {
                    "Default": {
                        "PoolId": "ourUserPoolId",
                        "AppClientId": "ourAppClientId",
                        "Region": "ourRegion"
                    }
                },
                "Auth": {
                    "Default": {
                        "authenticationFlowType": "USER_SRP_AUTH",
                        "socialProviders": [],
                        "usernameAttributes": [
                            "EMAIL"
                        ],
                        "signupAttributes": [
                            "EMAIL"
                        ],
                        "passwordProtectionSettings": {
                            "passwordPolicyMinLength": 8,
                            "passwordPolicyCharacters": [
                                "REQUIRES_LOWERCASE",
                                "REQUIRES_UPPERCASE",
                                "REQUIRES_NUMBERS",
                                "REQUIRES_SYMBOLS"
                            ]
                        },
                        "mfaConfiguration": "OFF",
                        "mfaTypes": [],
                        "verificationMechanisms": [
                            "EMAIL"
                        ]
                    }
                }
            }
        }
    }
}''';

**Comment by dnys1** [comment]
Hi @oantacamelia, sorry you are facing this issue. Would you be able to try on the latest version of the library and let me know if you still experience problems? This issue has been fixed in newer versions.

```yaml
amplify_auth_cognito: ^1.0.0-next.1
amplify_flutter: ^1.0.0-next.1
```

**Comment by radumunteadevm8** [comment]
I added the new version of amplify but now when I try to get the user token it throws an exception. Is it necessary to have an identity pool in order to have access to the isToken?

**Comment by oantacamelia** [comment]
I added the new version of amplify but now when I try to get the user token it throws an exception. Is it necessary to have an identity pool in order to have access to the isToken?

```
 final result = await Amplify.Auth.fetchAuthSession(
        options: CognitoSessionOptions(getAWSCredentials: true));
    final cognitoAuthSession = (result as CognitoAuthSession);
   var token =  cognitoAuthSession.userPoolTokens?.idToken.raw;
```
`I/flutter ( 2169): ERROR | FetchAuthSessionStateMachine | Emitted error: InvalidAccountTypeException(message: No identity pool registered for this account, recoverySuggestion: Register an identity pool using the CLI or set getAWSCredentials to false, underlyingException: null)
I/flutter ( 2169): #0      FetchAuthSessionStateMachine.onFetchAuthSession (package:amplify_auth_cognito_dart/src/state/machines/fetch_auth_session_state_machine.dart:205:7)
I/flutter ( 2169): <asynchronous suspension>
I/flutter ( 2169): #1      FetchAuthSessionStateMachineBase.resolve (package:amplify_auth_cognito_dart/src/state/machines/generated/fetch_auth_session_state_machine_base.dart:37:9)
I/flutter ( 2169): <asynchronous suspension>
I/flutter ( 2169): #2      StateMachine._listenForEvents (package:amplify_core/src/state_machine/state_machine.dart:171:9)
I/flutter ( 2169): <asynchronous suspension>
I/flutter ( 2169): 
`

**Comment by dnys1** [comment]
Ah apologies for the confusion. In vNext, setting `getAWSCredentials: true` will result in an exception when no identity pool is present. This is different from v0. To fix the error, simply remove the `getAwsCredentials: true` option and it should work.

**Comment by oantacamelia** [comment]
Thank you @dnys1 for the quick response. I've tested, as you suggested and it still returns an expired token. So, I just logged in as usual and at first returns a valid token, but when this expires doesn't return a new one.

**Comment by oantacamelia** [comment]
What I see to work for now is using this option, which refreshes the token every time.
`await Amplify.Auth.fetchAuthSession(options: const CognitoSessionOptions(forceRefresh: true));`

So we saved the last token locally and just made the request for another one, only when it is actually expired.

**Comment by ragingsquirrel3** [comment]
@oantacamelia ^1.0.0-next.1 is still in dev preview and not yet ready for production usage. We will send out updates when we have a GA, production-ready release.

**Comment by oantacamelia** [comment]
@ragingsquirrel3 Can another solution be found in the meantime? This is a blocker for us since we cannot release the app.

**Comment by slimkrazy** [comment]
Hey @ragingsquirrel3,

Is there an ETA for this release?

**Comment by vgribok** [comment]
> Hi @oantacamelia, sorry you are facing this issue. Would you be able to try on the latest version of the library and let me know if you still experience problems? This issue has been fixed in newer versions.
> 
> ```yaml
> amplify_auth_cognito: ^1.0.0-next.1
> amplify_flutter: ^1.0.0-next.1
> ```

I am not sure whether this issue was supposed to be fixed in the dev preview, but with the
```yaml
  amplify_flutter: ^1.0.0-next.6
  amplify_auth_cognito: ^1.0.0-next.6
```
it is still a problem (tested on the Android emulator). The auth session has `isSignedIn` set to true, but the access token and the identity token are both expired many hours ago. The emulator is online and has an internet connection. After logging off and logging back in the tokens got refreshed.

**Comment by dnys1** [comment]
Hi @vgribok - thanks for the update. I am not aware of what may be causing this issue, but would like to investigate more. Is the reproduction something like this?

1. Sign in with username/password
2. Without closing app, shutdown emulator so that it saves state.
3. After enough time has elapsed such that the tokens are expired, hot boot the emulator to the saved state.
4. Call `fetchAuthSession` and receive an expired an object which has `isSignedIn=true` and expired ID/access tokens.

**Comment by vgribok** [comment]
Yes, @dnys1, the steps you've listed are correct. This issue sounds like it could be an Android emulator issue, or an interplay of the libraries and emulator, because I think I have not encountered this issue on iOS, or on the physical Android device. Weird issues start cropping up after running Android emulator for a while, and they go away after the emulator is cold-rebooted.

**Comment by Jordan-Nelson** [comment]
This issue should be addressed in v1.0.0 of Amplify Flutter, which is now stable. Would you please upgrade to the latest version and let us know if you can still reproduce this issue?

This release also includes web and desktop support for Auth, API, Analytics, and Storage. You can see the list of new features and bug fixes in the [release notes](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.0.0), and see more details on how to migrate in the [upgrade guide](https://docs.amplify.aws/lib/project-setup/upgrade-guide/q/platform/flutter/).

**Comment by Jordan-Nelson** [comment]
I am going to close this issue out. If anyone is seeing this issue on v1.0.0 or later, please let us know and we can re-open the issue.
